### PR TITLE
update of bo usage count and size

### PR DESCRIPTION
**Pull request by sarab96**
#### Problem solved by the commit
This is following change to update bo usage count & size

#### Risks (if any) associated the changes in the commit
Following PR swill use this change to show in sysfs and xbutil

#### What has been tested and how, request additional testing if necessary
Tested locally

#### Documentation impact (if any)
Will be documented after updates in xbutil

### CODE REVIEW

**Comment by xilinxgitops** [comment]
retest this please

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.h [hunk 1]
```diff
@@ -149,6 +149,32 @@ static inline unsigned xocl_bo_type(unsigned user_flags)
...
+		idx = XOCL_BO_USAGE_DEV_ONLY;
+	case XOCL_BO_IMPORT:
+		idx = XOCL_BO_USAGE_IMPORT;
+	case XOCL_BO_EXECBUF:
+		idx = XOCL_BO_USAGE_EXECBUF;
+	case XOCL_BO_CMA:
+		idx = XOCL_BO_USAGE_CMA;
+	default:
+		idx = -1;
+	}
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.h:173]: you don't need to "break"?

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c [hunk 2]
```diff
@@ -658,6 +658,19 @@ void xocl_mm_update_usage_stat(struct xocl_drm *drm_p, u32 ddr,
 	drm_p->mm_usage_stat[ddr]->bo_count += count;
 }
 
+int xocl_bo_update_usage_stat(struct xocl_drm *drm_p, int bo_idx,
```
Comment (maxzhen) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:661]: from the places where this function is called, you always convert bo flags to index, then pass it into this function here. why you can't use bo flags directly as input and find out the index internally? or do you expect the caller of this function may not know the bo flags in some cases?

**Review by maxzhen — CHANGES_REQUESTED** [review]

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.h (hunk 1 above)
Comment (sarab96) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.h]: updated

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c (hunk 2 above)
Comment (sarab96) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:661]: bo types are defined in xocl_bo.h and may not be good to inclide this file in xocl_drm

Comment (maxzhen) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:661]: why you can't include it? it is just a header...

Comment (sarab96) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c]: Moved implementation to xocl.bo.c This is best

**Review by maxzhen — APPROVED** [review]

**Comment by dayeh-xilinx** [comment]
retest this please. Disabled PCIE emulation test on Ubuntu.

**Comment by xilinxgitops** [comment]
retest this please

**Pull request merged**